util/election.go

```diff
@@ -29,6 +29,8 @@ type MasterElection interface {
 	ResignAndRestart(context.Context) error
 	// Close permanently stops the mastership election process.
 	Close(context.Context) error
+	// GetCurrentMaster returns the instance ID of the current elected master, if any.
```

Comment:
Maybe add a little bit of text about "instance ID" as it's new to this interface -- is it unique and stable and therefore safe to use for election-related behaviour, or is it more of an indication that's a debugging help but not to be relied on for serious distributed system design?

Reply:
Also, maybe have a specific documented error value that the interface can use to indicate no-leader (and map `concurrency.ErrElectionNoLeader` to it in the etcd code)?

Reply:
done x2